src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/brgemm_blocking.cpp

```diff
@@ -74,89 +90,101 @@ bool BrgemmBlocking::run(LinearIR& linear_ir, LinearIR::constExprIt begin, Linea
+                 entries.emplace_back(copy_b_expr->get_input_port(0), true);
+            } else {
+                 entries.emplace_back(brgemm_expr->get_input_port(1), true);
+                if (brgemm->is_with_compensations())
+                    entries.emplace_back(brgemm_expr->get_input_port(2), true);
             }
 
-            *in_1_subtensor.rbegin() = block_size_n;
-            *out_subtensor.rbegin() = block_size_n;
-            auto loop_begin_it = expr_it, loop_end_it = std::next(expr_it);
-            std::vector<LoopPort> entries{LoopPort(brgemm_expr->get_input_port(0), false),
-                                          LoopPort(brgemm_expr->get_input_port(1), true)};
-            if (brgemm->is_with_compensations()) {
-                entries.emplace_back(brgemm_expr->get_input_port(2), true);
-            } else if (brgemm->is_amx()) {
-                move_new_memory_buffer(linear_ir, expr_it);
-                loop_begin_it = std::prev(expr_it);
-            }
             std::vector<LoopPort> exits{LoopPort(brgemm_expr->get_output_port(0), true)};
             loop_manager->mark_loop(loop_begin_it, loop_end_it, n, block_size_n, 0, entries, exits);
         };
 
-        auto apply_k_blocking = [&]() {
-            const auto& k = *in_0_planar_dims.rbegin();
-            OPENVINO_ASSERT(k == *(in_1_planar_dims.rbegin() + 1), "Brgemm input descriptors have different K dimension value.");
-            const auto block_size_k = brgemm->get_k_block_size();
-            if (block_size_k >= k) {
-                *in_0_subtensor.rbegin() = k;
-                *(in_1_subtensor.rbegin() + 1) = k;
-                brgemm->set_beta(0.f);
-                return;
-            }
-
-            *in_0_subtensor.rbegin() = block_size_k;
-            *(in_1_subtensor.rbegin() + 1) = block_size_k;
-            auto loop_begin_it = expr_it, loop_end_it = std::next(expr_it);
-            std::vector<LoopPort> entries{LoopPort(brgemm_expr->get_input_port(0), true, 0),
-                                          LoopPort(brgemm_expr->get_input_port(1), true, 1)};
-            if (brgemm->is_with_compensations()) {
-                entries.emplace_back(brgemm_expr->get_input_port(2), false, 1);
-            } else if (brgemm->is_amx()) {
-                move_new_memory_buffer(linear_ir, expr_it);
-                loop_begin_it = std::prev(expr_it);
+        auto mark_k_blocking = [&]() {
+            const auto loop_begin_it = get_loop_begin_pos(linear_ir, expr_it);
+            const auto loop_end_it = std::next(expr_it);
+
+            std::vector<LoopPort> entries{LoopPort(brgemm_expr->get_input_port(0), true, 0)};
+            if (brgemm->is_with_data_repacking()) {
+                 entries.emplace_back(copy_b_expr->get_input_port(0), true, 1);
+            } else {
+                 entries.emplace_back(brgemm_expr->get_input_port(1), true, 1);
+                if (brgemm->is_with_compensations())
```

Comment:
Perfect finding, thank you: it seems like I forgot to remove this code during final refactoring.
At the moment, in case of `brgemm` with compensations, we mark all loops starting from `copy_b` expression, so we can just put 0's input port of `copy_b` in `entries`: no special handling for `brgemm` with compensations is needed